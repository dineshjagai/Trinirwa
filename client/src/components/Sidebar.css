.sidebar {
    background-color: white;
    border-radius: 8px 0px 8px 0px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25); ;
    display: inline-block;
    width: 400px;
}
.img_box {
    float: left;
    padding: 4px;
}
.username_box {
    float: left;
    padding-left: 4px;
}
.header_box {
    display: flex;
    align-items: center;
    width: 100%;
}
#img {
    width: 80px;
    height: 80px;
    border-radius: 80px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.user {
    font-size: large;
    font-weight: bold;
    color: #0C8367;
}

span :hover {
    background-color: aqua;
}

.button_home {
    width: 100px;
}





