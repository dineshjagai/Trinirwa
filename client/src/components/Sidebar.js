/* eslint-disable react/no-array-index-key */
import React from 'react';
import * as FaIcons from 'react-icons/fa';
import { Route, BrowserRouter as Router, Link } from 'react-router-dom';
import './Sidebar.css';
import ProfilePage from './ProfilePage';

const SideBarData = [
  {
    title: 'Followers',
    path: '/followers',
    icon: <FaIcons.FaUserFriends />,
    cName: 'nav-text',
  },
];
function reload() {
  window.location.reload();
}

function SideBar({ uid }) {
  const profile = `/profile/${uid}`;
  return (
    <Router>
      <div className="sidebar">
        <ul className="sidebar-items">
          <li className="nav-text">
            <Link to={profile}>
              <FaIcons.FaUserFriends />
              {/* get username from uid
              <span>{username}</span> */}
            </Link>
          </li>
          {SideBarData.map((item, index) => (
            <li key={index} className={item.cName}>
              <Link onClick={reload} to={item.path}>
                {item.icon}
                <span>{item.title}</span>
              </Link>
            </li>
          ))}
        </ul>
      </div>
      <Route path="/profile" component={() => <ProfilePage />} />
    </Router>

  );
}

export default SideBar;
