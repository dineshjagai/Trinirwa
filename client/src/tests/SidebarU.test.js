import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Sidebar from '../components/SidebarU';
import '@testing-library/jest-dom';
import { act } from 'react-dom/test-utils';

//    beforeEach(() => {
// jest.mock('react-router-dom', () => ({
//      useHistory: () => ({
//         push: jest.fn(),
//      }),
//     }));
//     });

test('Test SidebarU snapshot', () => {
  const component = renderer.create(<Sidebar />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('Test SidebarU button', () => {
  const { getByRole } = render(<Sidebar />);
  const btn = document.getElementsByClassName('button_profile')[0];
  expect(btn).toBeInTheDocument();
});


