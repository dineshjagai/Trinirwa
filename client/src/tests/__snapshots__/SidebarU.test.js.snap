// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SidebarU snapshot 1`] = `
<div
  className="sidebar"
>
  <div
    className="bigButton"
    style={
      Object {
        "marginBottom": "5px",
      }
    }
  >
    <button
      className="button_profile"
      onClick={[Function]}
      type="button"
    >
      <div
        className="header_box"
      >
        <div
          className="img_box"
        >
          <img
            alt=""
            id="img"
            src="/api/viewfile/"
          />
        </div>
        <div
          className="username_box"
        >
          <span
            className="user"
          />
        </div>
      </div>
    </button>
  </div>
  <div
    style={
      Object {
        "marginBottom": "5px",
      }
    }
  >
    <div
      className="box-container"
    >
      <div
        className="buttonFollowers"
      >
        <div>
          <button
            className="btn"
            id="delete"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "border": "none",
                "color": "#00695c",
              }
            }
            type="button"
          >
            All friends
          </button>
        </div>
      </div>
      <div
        className="title"
        style={
          Object {
            "color": "#0C8367",
          }
        }
      >
        Friends
      </div>
      <br />
      <div
        className="box"
      >
        <div
          className="containers"
        />
      </div>
    </div>
  </div>
</div>
`;
